### PR TITLE
Rename Mmlogic to Queryservice

### LINKED ISSUE #996: API change: Move AssignTickets to the FE, rename services
## Overview

Move Backend.AssignTickets to the grpc service Frontend (TicketApi).

Rename the following:
Backend -> MatchApi
Frontend -> TicketApi
MmLogic -> QueryApi

## Motivation

We've known for awhile that these names were bad, but weren't sure what better ones were.  Lets fix that before going much further.

The move of AssignTickets is also to rationalize what the endpoints are doing a little bit.  The original inception was that different pieces of the game architecture would talk to specific pieces of OM.  So the frontend wasn't the "frontend", but instead is the API surface that the frontend talk to.  This has proved to not be intuitive, as well as confusing when we've tried to start talking about the /actual/ frontend component.

This split aligns much more closely with what the endpoints are actually doing:
MatchApi = calling the MMF and synchronizer, doing batch processing.  Large long lived requests.
TicketApi = Modifications and gets (by id) to tickets.  Small short term requests modifying redis.
QueryApi = Getting bulk data of or about tickets.  Likely to have caching and other optimizations in the future.  Very large streaming requests.

## Impact

The director type components will need an additional connection to the TicketApi, where they previously would only have a connection to the Backend (MatchApi).  Some renaming in client libraries required as well.

## Change Proto

Following the guidelines at: https://cloud.google.com/apis/design/naming_convention#interface_names

The names are appended with Api, as Match, Ticket, and Query are all names used elsewhere in the proto definitions.

```proto
// The QueryApi service implements helper APIs for Match Function to query Tickets from state storage.
service QueryApi {
```

```proto
// The MatchApi service implements APIs to generate matches and handle ticket assignments.
service MatchApi {
```

```proto
// The TicketApi service implements APIs to manage and query status of a Tickets.
service TicketApi {
```

### PR DESCRIPTION
**Pull request by yfei1**
Resolved #996.

1. Ran the following command and renamed the file content with name that contains `mmlogic` to `queryservice` instead.
```
find . -type f -not -path "./.git/*" -exec sed -i 's/Mmlogic/Queryservice/g' {} +
find . -type f -not -path "./.git/*" -exec sed -i 's/mmlogic/queryservice/g' {} +
find . -type f -not -path "./.git/*" -exec sed -i 's/MmLogic/QueryService/g' {} +
```

2. Manually rename the file name under `internal/app/mmlogic` and `cmd/mmlogic` from `mmlogic.go` to `query.go` to keep the image name consistent with our backend and frontend naming.

TODO: Rename backend and frontend API to `BackendService` and `FrontendService` instead.

### CODE REVIEW

**Comment by Laremere** [comment]
I think "QueryService" as a name for the image/deployment/etc is a bit much, and introduces inconsistancy with "Frotend" and "Backend".

As separate PRs:
- Rename the grpc services to "FrontendService", "BackendService", and "QueryService".
- Rename the image and associated files to "query"

**Comment by yfei1** [comment]
Done. Renamed the `Mmlogic` file to `Query` so that the generated image names are now `om-query` instead. Also updated the PR description.

File: internal/app/query/query_service.go [hunk 1]
```diff
@@ -29,13 +29,13 @@ import (
...
-		"component": "app.mmlogic",
+		"component": "app.query",
 	})
 )
 
-// The MMLogic API provides utility functions for common MMF functionality such
+// The queryservice API provides utility functions for common MMF functionality such
 // as retreiving Tickets from state storage.
-type mmlogicService struct {
+type queryserviceService struct {
```
Comment (Laremere) [inline internal/app/query/query_service.go]: Why service service?  Just queryService?

File: pkg/harness/function/golang/matchfunction_service.go [hunk 2]
```diff
@@ -47,9 +47,9 @@ type MatchFunction func(*MatchFunctionParams) ([]*pb.Match, error)
 // matchFunctionService implements pb.MatchFunctionServer, the server generated
 // by compiling the protobuf, by fulfilling the pb.MatchFunctionServer interface.
 type matchFunctionService struct {
-	cfg           config.View
-	function      MatchFunction
-	mmlogicClient pb.MmLogicClient
+	cfg                config.View
+	function           MatchFunction
+	queryserviceClient pb.QueryServiceClient
```
Comment (Laremere) [inline pkg/harness/function/golang/matchfunction_service.go]: Nit: queryServiceClient

File: pkg/matchfunction/matchfunction.go [hunk 3]
```diff
@@ -23,11 +23,11 @@ import (
...
 )
 
-// QueryPool queries mmlogic and returns the tickets that belong to the specified pool.
-func QueryPool(ctx context.Context, mml pb.MmLogicClient, pool *pb.Pool) ([]*pb.Ticket, error) {
+// QueryPool queries queryservice and returns the tickets that belong to the specified pool.
+func QueryPool(ctx context.Context, mml pb.QueryServiceClient, pool *pb.Pool) ([]*pb.Ticket, error) {
 	query, err := mml.QueryTickets(ctx, &pb.QueryTicketsRequest{Pool: pool})
 	if err != nil {
-		return nil, fmt.Errorf("Error calling mmlogic.QueryTickets: %w", err)
+		return nil, fmt.Errorf("Error calling queryservice.QueryTickets: %w", err)
```
Comment (Laremere) [inline pkg/matchfunction/matchfunction.go]: queryService

File: tutorials/custom_evaluator/matchfunction/main.go [hunk 4]
```diff
@@ -24,14 +24,14 @@ import (
...
 
 // This tutorial implenents a basic Match Function that is hosted in the below
-// configured port. You can also configure the Open Match MMLogic service endpoint
+// configured port. You can also configure the Open Match QueryService endpoint
 // with which the Match Function communicates to query the Tickets.
 
 const (
-	mmlogicAddress = "om-mmlogic.open-match.svc.cluster.local:50503" // Address of the MMLogic service endpoint.
-	serverPort     = 50502                                           // The port for hosting the Match Function.
+	queryserviceAddress = "om-query.open-match.svc.cluster.local:50503" // Address of the QueryService endpoint.
```
Comment (Laremere) [inline tutorials/custom_evaluator/matchfunction/main.go]: queryService

**Comment by yfei1** [comment]
> There's a bunch of places which need proper camel caps. I commented on some, but please fix all.

Done.

File: examples/scale/scenarios/util.go [hunk 5]
```diff
@@ -91,17 +91,17 @@ func (e StatProcessor) Log(w io.Writer) {
 	})
 }
 
-func getMmlogicGRPCClient() pb.MmLogicClient {
-	conn, err := grpc.Dial(mmlogicAddress, testing.NewGRPCDialOptions(logger)...)
+func getQueryserviceGRPCClient() pb.QueryServiceClient {
```
Comment (Laremere) [inline examples/scale/scenarios/util.go]: correct casing

**Pull request merged**